lock-api/src/main/java/com/palantir/lock/client/timestampleases/NamespacedTimestampLeaseServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client.timestampleases;
+
+import com.palantir.atlasdb.timelock.api.GetMinLeasedTimestampRequests;
+import com.palantir.atlasdb.timelock.api.GetMinLeasedTimestampResponses;
+import com.palantir.atlasdb.timelock.api.Namespace;
+import com.palantir.atlasdb.timelock.api.NamespaceTimestampLeaseRequest;
+import com.palantir.atlasdb.timelock.api.NamespaceTimestampLeaseResponse;
+import com.palantir.lock.client.InternalMultiClientConjureTimelockService;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import java.util.Map;
+import java.util.function.Supplier;
+
+final class NamespacedTimestampLeaseServiceImpl implements NamespacedTimestampLeaseService {
+    private final Namespace namespace;
+    private final InternalMultiClientConjureTimelockService delegate;
+
+    NamespacedTimestampLeaseServiceImpl(Namespace namespace, InternalMultiClientConjureTimelockService delegate) {
+        this.namespace = namespace;
+        this.delegate = delegate;
+    }
+
+    @Override
+    public NamespaceTimestampLeaseResponse acquireTimestampLeases(NamespaceTimestampLeaseRequest request) {
+        Map<Namespace, NamespaceTimestampLeaseResponse> response =
+                delegate.acquireTimestampLeases(Map.of(namespace, request));
+        validateResponseIsCoherent(response, () -> SafeArg.of("request", request));
+        return response.get(namespace);
+    }
+
+    @Override
+    public GetMinLeasedTimestampResponses getMinLeasedTimestamps(GetMinLeasedTimestampRequests request) {
+        Map<Namespace, GetMinLeasedTimestampResponses> response =
+                delegate.getMinLeasedTimestamps(Map.of(namespace, request));
+        validateResponseIsCoherent(response, () -> SafeArg.of("request", request));
+        return response.get(namespace);
+    }
+
+    private void validateResponseIsCoherent(Map<Namespace, ?> response, Supplier<SafeArg<?>> requestArg) {
```

Comment:
Can you do Object for the requestArg, instead of Supplier?

Reply:
Or honestly, ignore good error hygiene, this should never happen

Reply:
To clarify: it's ok for you to do the check, if you think it's valuable, but don't get out of your way to give a good error message. This should never happen.